### DISCUSSION TITLE
How can I continue an unfinished message?

### DISCUSSION THREAD

**Discussion by Mashiro-Yukino** [brianpetro/obsidian-smart-connections#151] [Q&A]
I find it is hard to continue the unfinished message after breaking if I used my notes, anyway to continue the unfinished message?

**Comment by Mara-Li** [comment]
I have the same problem, many replies are unfinished : 
![image](https://user-images.githubusercontent.com/30244939/229379345-feba654d-0662-4c9b-b93b-648829b53a5c.png)

**Comment by brianpetro** [comment]
@Mashiro-Yukino @Lisandra-dev thanks for bringing this to my attention. 

When using your notes as context, often we use up all available space for text as context.  So, after this completion from the model, the context sourced from the notes is dropped to make room for additional messages. This is why continuing doesn't work after messages that used your notes as context.

There is one relevant improvement coming soon: the ability to change the size of the output. However, it's important to note, since the text size limit is a combination of both the input text and the output text, any increase in output will reduce the amount of context sourced from your notes.

To get around this using prompt engineering, you can try summarizing context from your notes, then making a subsequent query that will use the summary instead of your raw notes as context.

This problem will also be improved as the text-size limits of the underlying AI models continue to increase. GPT-4 does currently offer double the text-size (tokens) limit as GPT-3.5, and it was already announced that GPT-4 will offer eight times the limit in an upcoming release. 

Let me know if you have any questions, thoughts or other ideas about how we might address this and make Smart Connections a better assistant. 

Brian 🌴

**Comment by Mara-Li** [comment]
It will be cool to have a text/symbole allow to cut the notes send in the prompt, in this case. 

For example `%% stop %%` or `%%---%%`

Also I think the plugin can remove the frontmatter from note entry

**Comment by brianpetro** [comment]
@Lisandra-dev interesting. Let me give that some thought and get back to you.

Thanks for the PR,
Brian 🌴

**Comment by Mara-Li** [comment]
Also I discover that using th word `continue` allow GPT to continue the unfinished messages!

**Comment by Mashiro-Yukino** [comment]
Yes, I tried to use "continue". But subsequent responses won't be based on your notes any more

**Comment by brianpetro** [comment]
> Yes, I tried to use "continue". But subsequent responses won't be based on your notes any more

@Mashiro-Yukino yes, this is true.

This is the relevant explanation:

> When using your notes as context, often we use up all available space for text as context. So, after this completion from the model, the context sourced from the notes is dropped to make room for additional messages. This is why continuing doesn't work after messages that used your notes as context.

https://github.com/brianpetro/obsidian-smart-connections/issues/142#issuecomment-1493491006 explains more about the issue, possible workarounds, and upcoming changes that will improve it.
